### PR TITLE
Enable HTTP/2

### PR DESCRIPTION
**Pull request by k1ng440**
Related to https://github.com/quick-lint/quick-lint-js/issues/750

* Switched to mpm_event from mpm_prefork
* Switched to php-fpm from mod_php

### CODE REVIEW

**Comment by k1ng440** [comment]
@strager please review it whenever you have time.

File: website/public/phpinfo.php [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<?php 
+
+phpinfo();
```
Comment (strager) [inline website/public/phpinfo.php:3]: Must fix: This is a security issue. Delete.

We shouldn't even be serving PHP on https://quick-lint-js.com/. (But that's not your patch's fault.)

File: infrastructure/quick-lint-js-web/apache2/apache2.conf [hunk 2]
```diff
@@ -220,6 +220,14 @@ CustomLog ${APACHE_LOG_DIR}/matomo_analytics.log matomo_analytics
 
 FileETag INode MTime
 
+# FCGI
+AddHandler fcgid-script .fcgi .php .fpl
```
Comment (strager) [inline infrastructure/quick-lint-js-web/apache2/apache2.conf:224]: We only need .php, right?

File: infrastructure/quick-lint-js-web/run-apache2.sh [hunk 3]
```diff
@@ -9,7 +9,11 @@ set -u
 printf '\n============================================================\n'
 printf '\nrunning web server: http://quick-lint-js.com%s/\n' "${DEV_TLD}"
 printf '\n============================================================\n'
-exec apache2ctl -D FOREGROUND
+
+php-fpm8.1 -c /etc/php/8.1/fpm/php-fpm.conf -F &
+apache2ctl -D FOREGROUND &
+
+wait
```
Comment (strager) [inline infrastructure/quick-lint-js-web/run-apache2.sh:16]: If we want `wait -n`, we should install Bash in the container and use it for this script. It looks like POSIX sh doesn't support `-n` or an equivalent. =\

File: infrastructure/quick-lint-js-web/apache2/mods-available/mpm_event.conf [hunk 4]
```diff
@@ -6,7 +6,7 @@
 # MaxRequestWorkers: maximum number of worker threads
 # MaxConnectionsPerChild: maximum number of requests a server process serves
 <IfModule mpm_event_module>
-	StartServers			 2
+	StartServers		 2
```
Comment (strager) [inline infrastructure/quick-lint-js-web/apache2/mods-available/mpm_event.conf:9]: Nit: Undo unnecessary whitespace changes like this.

File: infrastructure/quick-lint-js-web/apache2/apache2.conf (hunk 2 above)
Comment (strager) [inline infrastructure/quick-lint-js-web/apache2/apache2.conf:224]: Do we need this directive at all? We already have a similar directive in `fcgid.conf`.

File: infrastructure/quick-lint-js-web/apache2/mods-available/fcgid.conf [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<IfModule mod_fcgid.c>
```
Comment (strager) [inline infrastructure/quick-lint-js-web/apache2/mods-available/fcgid.conf:1]: Is this file used? I don't see an `a2enconf fcgid`.

**Review by strager — CHANGES_REQUESTED** [review]

File: infrastructure/quick-lint-js-web/apache2/mods-available/fcgid.conf (hunk 5 above)
Comment (k1ng440) [reply infrastructure/quick-lint-js-web/apache2/mods-available/fcgid.conf:1]: Yes, it's being used. we are overwriting the file.

File: infrastructure/quick-lint-js-web/run-apache2.sh (hunk 3 above)
Comment (k1ng440) [reply infrastructure/quick-lint-js-web/run-apache2.sh]: switched to bash and had to use pids with wait -n

File: website/public/phpinfo.php (hunk 1 above)
Comment (k1ng440) [reply website/public/phpinfo.php]: done

**Comment by strager** [comment]
I deployed this PR, and HTTP/2 seems to be working.

However, our [PHP app](https://admin.quick-lint-js.com/matomo/) is broken:
>  Deprecated: DateTime::__construct(): Passing null to parameter #1 ($datetime) of type string is deprecated in /var/www/admin.quick-lint-js.com/matomo/vendor/monolog/monolog/src/Monolog/Logger.php on line 324

I think a Matomo upgrade will fix it: https://github.com/matomo-org/matomo/issues/17686

**Comment by strager** [comment]
I rolled back and Matomo is still broken. Hmm. Maybe it was always broken. :zany_face:

**Comment by strager** [comment]
It looks like Matomo has been broken since June 8 (i.e. when I switched to the Docker container). I upgraded to Matomo 4.9.1 the UI seems to be working. (Logging is probably broken still.)

**Review by strager — APPROVED** [review]
